File: cgi-bin/list.pl
#!/usr/bin/perl -w
# list.pl - display the survey results

use CGI ":standard";

# error - a function to produce an error message for the client
#         and exit in case of open errors

sub error {
    print "Error - file could not be opened in list.pl <br/>";
    print end_html();
    exit(1);
}

# Begin main program
# Initialize file locking/unlocking parameter

$LOCK = 2;
$UNLOCK = 8;

print header(
    -charset=>"UTF-8",
);

# Open, lock, read, and unlock the survey data file

open(SURVDAT, "<survdat.dat") or error();
flock(SURVDAT, $LOCK);
@data = <SURVDAT>;
flock(SURVDAT, $UNLOCK);
close(SURVDAT);

# Create the beginning of the result Web page

print start_html(
    -title=>"Results",
    -style=>{src=>'../css/index.css'},
);
print "<h2>Results of the Basic Information Questionnaire</h2>";

# Make the column titles array

@col_titles = ("Name", "Age", "Gender", "E-mail", "Checkbox");

# Create the column titles in HTML by giving their address to the th
#  function and storing the return value in the @rows array

@rows = th(\@col_titles);

my $index = 0;

foreach (@data) {
    @item = split(/,/);
    my $checkbox = sprintf("<input type='checkbox' value='%d' name='ids'/>", ++$index);
    push @item, $checkbox;
    push(@rows, td(\@item));
}

# Create the table for the female survey results
#  The address of the array of row addresses is passed to Tr

print '<form action="delete.pl" method="post">';
print table(caption("Questionnaire Data"), Tr(\@rows));
print '<br/>';
print scalar @data,' item(s) in total. To delete selected items, click ';
print '<input type="submit" value="this button"/>';
print '.';
print '</form>';
print '<br></br>[<a href="../hw/unit3/index.html">Return Unit 3</a>]&nbsp;&nbsp;[<a href="list.pl">Refresh</a>]';

print end_html();

File: php/guest/list.php
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <title>Guest Book</title>
    <link href="../../css/index.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<?php
require_once '../database.php';

if ($result = mysqli_query($link, "SELECT * FROM `guest`")) {
    print "<h2>Results of the Basic Information Questionnaire</h2>";
    $cnt = mysqli_num_rows($result);
    if ($cnt == 0) {
        print "No result found.\n";
    } else {
        print "<form action=\"delete.php\" method=\"post\">";
        print "<table><caption>Guest Book</caption>\n";
        print "<tr><th>ID</th> <th>Name</th> <th>Age</th> <th>Gender</th> <th>E-mail</th> <th>Checkbox</th></tr>\n";
        while ($row = mysqli_fetch_assoc($result)) {
            echo "<tr>";
            echo "<td>" . $row['Guest_ID'] . "</td>";
            echo "<td>" . $row['Guest_Name'] . "</td>";
            echo "<td>" . $row['Age'] . "</td>";
            echo "<td>" . $row['Gender'] . "</td>";
            echo "<td>" . $row['E_mail'] . "</td>";
            echo "<td><input type='checkbox' value='{$row['Guest_ID']}' name='ids[]'/></td>";
            echo "</tr>\n";
        }
        print "</table>\n";
        print "<br>" . $cnt . " item(s) in total. To delete selected items, click <input type=\"submit\" value=\"this button\"/>.";
        print "</form>\n";
    }
} else {
    print "mysqli_query failed: ". mysqli_error($link);
}

mysqli_close($link);
?>
<br><br>[<a href="../../hw/unit4/index.html">Return Unit 4</a>]&nbsp;&nbsp;[<a href="list.php">Refresh</a>]
</body>
</html>